### PR TITLE
Course updates

### PR DESCRIPTION
**Pull request by ric9176**
1. This PR implements a new util function that can be used to achieve the following on instance pages by passing the training object down to the curriculum, visually resulting in the following: 
Screen Shot 2019-06-21 at 1 49 47 PM
(waiting on any code comments before applying this across all relevant instances)

2. Prices have been removed for trainings which are waitlist only: 
Screen Shot 2019-06-21 at 10 16 28 AM

3. All new pricing has been applied in UM to upcoming instances as discussed

4. Applies the same fix as used with React curriculums to solve the issue with displaying workshops of multiple types: 

Screen Shot 2019-06-21 at 4 22 40 PM

### CODE REVIEW

File: src/components/curriculum/CurriculumBootcamp.js [hunk 1]
```diff
@@ -43,22 +45,30 @@ const CurriculumBootcamp = ({
...
     <React.Fragment>
       <H4>Course outline:</H4>
       <Section
         {...commonProps}
         title="Evening pre-bootcamp (optional)"
         name="day0"
-        subTitle="React 101 and JS fundamentals"
+        subTitle={`React 101 and JS fundamentals (${TrainingDateByDay({
+          training,
+        })}  ${meetupTime})`}
```
Comment (alexlbr) [inline src/components/curriculum/CurriculumBootcamp.js]: @ric9176 the one in August should display 6:30 pm to 9 pm. what about:
```
// TODO remove this DEPRECATED_ logic after July 2019
const DEPRECATED_IS_JULY2019 = isJuly2019(startDate);
const meetupTime = DEPRECATED_IS_JULY2019?  '16:30 - 18:30': '6:30 - 9 pm'
```

File: src/components/utils/index.js [hunk 2]
```diff
@@ -130,14 +130,20 @@ export const DisplayComponentsUsingCss = styled(Components)`
...
+  utcOffset = 60,
+  format = "D MMM 'YYYY",
+  day = 0
+) => {
   const targetTime = new Date(utcDate),
     now = new Date(),
     minutesToMilliseconds = 60000
   const localOffsetInMs = now.getTimezoneOffset() * minutesToMilliseconds
   const utcOffsetInMs = utcOffset * minutesToMilliseconds
+  const dayOffset = day * 1440 * minutesToMilliseconds
```
Comment (alexlbr) [inline src/components/utils/index.js]: @ric9176 what is 1440? I'd create a const for that number to give meaning, as we did with `minutesToMilliseconds = 60000`. Maybe `const dayOffset` is `const dayOffsetInMs`?

File: src/components/curriculum/CurriculumBootcamp.js [hunk 3]
```diff
@@ -34,6 +35,7 @@ const CurriculumBootcamp = ({
   showLinkToCurriculum = true,
   trainings,
   showHackathon = false,
+  training = {},
```
[the remark this answers is not in the document]
Comment (ric9176) [reply src/components/curriculum/CurriculumBootcamp.js]: Sure thing I’ll refactor this, the reason that occurred to me to pass the whole thing was while I was doing it I realised that in future the curriculum text should be coming via the training object (not hardcoded) so I passed the whole thing bc I see that as a future implementation. It’s true tho for now we just need start date and nothing else.

File: src/components/curriculum/CurriculumBootcamp.js (hunk 1 above)
Comment (ric9176) [reply src/components/curriculum/CurriculumBootcamp.js]: 👍

File: src/components/utils/index.js [hunk 4]
```diff
@@ -130,14 +130,20 @@ export const DisplayComponentsUsingCss = styled(Components)`
         : ''}
 `
 
-export const formatUTC = (utcDate, utcOffset = 60, format = "D MMM 'YYYY") => {
+export const formatUTC = (
+  utcDate,
+  utcOffset = 60,
+  format = "D MMM 'YYYY",
+  day = 0
```
[the remark this answers is not in the document]
Comment (ric9176) [reply src/components/utils/index.js]: 👍 will update to offsetDay, to me it’s the idea that it reads day:1 so when it’s used you ask for the date of the first day (or don’t pass the arg for normal start date) , that’s true tho inside the function it’s strange, I’ll update to offsetDays

File: src/components/utils/index.js [hunk 5]
```diff
@@ -172,6 +178,11 @@ export const formatUTC = (utcDate, utcOffset = 60, format = "D MMM 'YYYY") => {
   }
 }
 
+export const TrainingDateByDay = ({ training = {}, day = 0 }) =>
```
[the remark this answers is not in the document]
Comment (ric9176) [reply src/components/utils/index.js]: I'll update naming, re date, we could do that, we'd also need to pass `training.utcOffset` directly in that case

**Pull request merged**